Description: Handle multiple files in genpeimg
Author: Stephen Kitt <skitt@debian.org>

genpeimg claims to process multiple files, but doesn't; this patch
fixes that.

--- a/mingw-w64-tools/genpeimg/src/genpeimg.c
+++ b/mingw-w64-tools/genpeimg/src/genpeimg.c
@@ -49,7 +49,6 @@
 unsigned short mask_pe_opt_hdr_dll_chara = 0xffff;
 unsigned short set_subsystem = 0xffff;
 int dump_information = 0;
-static char *file_name = NULL;
 
 static void __attribute__((noreturn))
 show_usage (void)
@@ -108,7 +107,6 @@
       if (h[0] != '-')
         {
 	  has_file = 1;
-	  file_name = h;
 	  continue;
 	}
       switch (h[1])
@@ -301,26 +299,32 @@
 
 int main (int argc, char **argv)
 {
+  char *file_name;
   pe_image *pe;
   --argc, ++argv;
   pass_args (argc, argv);
 
-  pe = peimg_load (file_name);
-  if (!pe)
-    {
-      fprintf (stderr, "File not found, or no PE-image\n");
-      return 0;
-    }
+  while (argc-- > 0) {
+    file_name = *argv++;
+    if (file_name[0] != '-' && file_name[0] != '+') {
+      pe = peimg_load (file_name);
+      if (!pe)
+      {
+        fprintf (stderr, "File %s not found, or not a PE-image\n", file_name);
+        return 0;
+      }
 
-  if (dump_information)
-    peimg_show (pe, stdout);
-  /* First we need to do actions which aren't modifying image's size.  */
-  peimg_set_hdr_characeristics (pe, set_pe_hdr_chara, mask_pe_hdr_chara);
-  peimg_set_hdr_opt_dll_characteristics (pe, set_pe_opt_hdr_dll_chara, mask_pe_opt_hdr_dll_chara);
-  if (set_subsystem != 0xffff)
-    peimg_set_hdr_opt_subsystem (pe, set_subsystem);
-  if (pe->pimg->is_modified)
-    pe->pimg->want_save = 1;
-  peimg_free (pe);
+      if (dump_information)
+        peimg_show (pe, stdout);
+      /* First we need to do actions which aren't modifying image's size.  */
+      peimg_set_hdr_characeristics (pe, set_pe_hdr_chara, mask_pe_hdr_chara);
+      peimg_set_hdr_opt_dll_characteristics (pe, set_pe_opt_hdr_dll_chara, mask_pe_opt_hdr_dll_chara);
+      if (set_subsystem != 0xffff)
+        peimg_set_hdr_opt_subsystem (pe, set_subsystem);
+      if (pe->pimg->is_modified)
+        pe->pimg->want_save = 1;
+      peimg_free (pe);
+    }
+  }
   return 0;
 }
